### PR TITLE
bug/17325 Unique upn validation fix

### PR DESCRIPTION
**Pull request by GeorgeChatzigiannis**
UPN validation should only trigger when id is different under the same UPN

### CODE REVIEW

File: admin/lib/validator/pupil-validator.js [hunk 1]
```diff
@@ -121,7 +121,8 @@ module.exports.validate = async (pupilData) => {
   // We need to check that the UPN is unique
   if (!(validationError.get('upn'))) {
     const existingPupil = await pupilDataService.findOne({ upn: pupilData.upn })
-    if (existingPupil) {
+    // if pupil is not stored already under the same id and UPN
```
Comment (jon-shipley) [inline admin/lib/validator/pupil-validator.js]: If I am reading this correctly UPN is not validated if pupil does not have an _id field.  Is there any chance of that happening, e.g. during create()

Comment (GeorgeChatzigiannis) [reply admin/lib/validator/pupil-validator.js]: The same validation is being used on both edit and create a pupil record.
Essentially this fixes the issue where editing a pupil would trigger validation to have an error due to getting the exact same user and thus disallowing the edit to go through.

Comment (GeorgeChatzigiannis) [reply admin/lib/validator/pupil-validator.js]: The change should only trigger validation if an existing upn is used from a pupil with different _id.

Comment (jon-shipley) [reply admin/lib/validator/pupil-validator.js]: Same issue?

**Pull request merged**